rasa/utils/tensorflow/model_data.py

```diff
@@ -139,6 +139,8 @@ def __reduce__(self) -> Tuple[Any, Any, Any]:
             A tuple.
         """
         pickled_state = super(FeatureArray, self).__reduce__()
+        if isinstance(pickled_state, str):
+            raise TypeError("np array __reduce__ returned string instead of tuple.")
```

Comment:
Apparently this can happen according to the type checker - but the documentation is non-existent so I wasn't able to figure out when this might be the case.